From: Debian CLI Applications Team <pkg-cli-apps-team@lists.alioth.debian.org>
Date: Fri, 20 May 2011 10:27:46 +0100
Subject: ubuntu_xdg-photo-dir

# ubuntu_xdg_photo_dir.patch: Patch by Andrew Higginson to set the default import target to
# XDG pictures dir/Photos (as translated). This only affects the first run, existing users
# will see no change
-- f-spot-0.6.0.0.orig/debian/patches/ubuntu_xdg_photo_dir.patch
++ f-spot-0.6.0.0/debian/patches/ubuntu_xdg_photo_dir.patch
@ -0,0 +1,16 @@
---
 src/Clients/MainApp/FSpot/Preferences.cs |    9 ++++++++-
 1 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/src/Clients/MainApp/FSpot/Preferences.cs b/src/Clients/MainApp/FSpot/Preferences.cs
index 5996b5f..676c1f5 100644
--- a/src/Clients/MainApp/FSpot/Preferences.cs
+++ b/src/Clients/MainApp/FSpot/Preferences.cs
@@ -155,7 +155,14 @@ namespace FSpot
 			case SCREENSAVER_DELAY:
 				return 4.0;
 			case STORAGE_PATH:
-				return System.IO.Path.Combine (Global.HomeDirectory, Catalog.GetString("Photos"));
+                String default_storage_path = System.IO.Path.Combine (
+                    Environment.GetFolderPath (Environment.SpecialFolder.MyPictures),
+                    Catalog.GetString ("Photos")
+                );
+				if (!System.IO.Directory.Exists (default_storage_path)) {
+					System.IO.Directory.CreateDirectory (default_storage_path);
+				}
+				return default_storage_path;
 			case EXPORT_EMAIL_SIZE:
 				return 3;	// medium size 640px
 			case EXPORT_EMAIL_ROTATE:
-- 
